### DISCUSSION TITLE
coverlet always runs dll in debug folder

### DISCUSSION THREAD

**Discussion by TerryChan2** [coverlet-coverage/coverlet#1455] [Q&A] [answered]
I have a test.dll in bin\Release folder but the bin\Debug folder is empty under a project folder. When I run coverlet as shown below

1) Go to project folder => cd C:\projects\project.tests
2) Run coverlet => coverlet .\bin\Release\test.dll --target "dotnet" --targetargs "test C:\projects\project.tests --no-build" --format cobertura

The output shows below:
_BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
Unable to instrument module: .\bin\Release\Interop.dll
System.NotSupportedException: Writing mixed-mode assemblies is not supported
   at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
   at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
   at Mono.Cecil.ModuleDefinition.Write(Stream stream, WriterParameters parameters)
   at Coverlet.Core.Instrumentation.Instrumenter.InstrumentModule() in /_/src/coverlet.core/Instrumentation/Instrumenter.cs:line 337
   at Coverlet.Core.Instrumentation.Instrumenter.Instrument() in /_/src/coverlet.core/Instrumentation/Instrumenter.cs:line 153
   at Coverlet.Core.Coverage.PrepareModules() in /_/src/coverlet.core/Coverage.cs:line 135
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
Test run for C:\projects\project.tests\bin\**Debug**\test.dll (.NETFramework,Version=v4.8)
Microsoft (R) Test Execution Command Line Tool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.
The test source file "C:\projects\project.tests\bin\**Debug**\test.dll" provided was not found._

If I copied the files from Release folder to Debug folder, the test cases can be executed as shown below:
_BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
Unable to instrument module: .\bin\Release\Interop.dll
System.NotSupportedException: Writing mixed-mode assemblies is not supported
   at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
   at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
   at Mono.Cecil.ModuleDefinition.Write(Stream stream, WriterParameters parameters)
   at Coverlet.Core.Instrumentation.Instrumenter.InstrumentModule() in /_/src/coverlet.core/Instrumentation/Instrumenter.cs:line 337
   at Coverlet.Core.Instrumentation.Instrumenter.Instrument() in /_/src/coverlet.core/Instrumentation/Instrumenter.cs:line 153
   at Coverlet.Core.Coverage.PrepareModules() in /_/src/coverlet.core/Coverage.cs:line 135
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
BadImageFormatException during MetadataReaderProvider.FromPortablePdbStream in InstrumentationHelper.PortablePdbHasLocalSource, unable to check if module has got local source.
Test run for C:\projects\project.tests\bin\**Debug**\test.dll (.NETFramework,Version=v4.8)
Microsoft (R) Test Execution Command Line Tool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Passed!  - Failed:     0, Passed:    15, Skipped:     0, Total:    15, Duration: 31 ms - test.dll (net48)_

How to config coverlet to run the dll in release folder?

**Comment by petli** [comment]
You have to tell `dotnet test` to run the release configuration:

```
coverlet .\bin\Release\test.dll --target "dotnet" --targetargs "test C:\projects\project.tests --no-build -c Release" --format cobertura
```

**Reply by TerryChan2** [reply]
Thanks petli.  It works!

I have another test2.dll and run the coverlet for that dll would produce below:

Test run for C:\projects\project2.tests\bin\Release\test2.dll (.NETFramework,Version=v4.5.2)
Microsoft (R) Test Execution Command Line Tool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Passed!  - Failed:     0, Passed:     7, Skipped:     0, Total:     7, Duration: 18 ms - test2.dll (net452)

Calculating coverage result...
  Generating report '.\result\coverage.cobertura.xml'
The path is empty. (Parameter 'path')

Is this issue related to https://github.com/coverlet-coverage/coverlet/issues/775 ?

**Reply by petli** [reply, marked as answer]
I can't say if it has anything to do with that issue. Maybe it is that the project uses .NET Framework 4.5.2, according to https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/DriversFeatures.md the minimum supported version is 4.6.1.
